File: src/game/entity/enemies/npcs/FlyingEnemy.java
package game.entity.enemies.npcs;

import game.entity.blocks.DestroyableBlock;
import game.entity.models.*;
import game.models.Coordinates;
import game.utils.Paths;

import java.util.HashSet;
import java.util.Set;

public abstract class FlyingEnemy extends IntelligentEnemy {
    public FlyingEnemy() {
        super();
    }

    public FlyingEnemy(Coordinates coordinates) {
        super(coordinates);
    }


    @Override
    public Set<Class<? extends Entity>> getObstacles() {
        Set<Class<? extends Entity>> baseObstacles = new HashSet<>(super.getObstacles());
        baseObstacles.remove(DestroyableBlock.class);
        return baseObstacles;
    }
}

File: src/game/entity/enemies/npcs/FlyingEnemy.kt
package game.entity.enemies.npcs

import game.entity.blocks.DestroyableBlock
import game.entity.models.Coordinates
import game.entity.models.Entity

/**
 * Enemy that can fly over destroyable blocks;
 */
abstract class FlyingEnemy : IntelligentEnemy {
    constructor() : super()
    constructor(coordinates: Coordinates?) : super(coordinates)
    constructor(id: Long) : super(id)

    override fun getObstacles(): Set<Class<out Entity>> = HashSet(super.getObstacles()).apply {
        remove(DestroyableBlock::class.java)
    }
}